File: app/src/main/java/org/autojs/autojs/tool/UiHandler.java
package org.autojs.autojs.tool;

import android.content.Context;
import android.os.Handler;
import android.os.Looper;

import org.autojs.autojs.util.ViewUtils;

/**
 * Created by Stardust on 2017/5/2.
 */
public class UiHandler extends Handler {

    private final Context mContext;

    public UiHandler(Context context) {
        super(Looper.getMainLooper());
        mContext = context;
    }

    public Context getContext() {
        return mContext;
    }

    public void toast(final String message) {
        post(() -> ViewUtils.showToast(mContext, message));
    }

    public void toast(final int resId) {
        post(() -> ViewUtils.showToast(mContext, resId));
    }

}

File: app/src/main/java/org/autojs/autojs/tool/UiHandler.kt
package org.autojs.autojs.tool

import android.content.Context
import android.os.Handler
import android.os.Looper
import android.widget.Toast
import android.widget.Toast.LENGTH_LONG
import android.widget.Toast.LENGTH_SHORT
import org.autojs.autojs.AutoJs
import org.autojs.autojs.runtime.ScriptRuntime
import org.autojs.autojs.runtime.api.ScriptToast
import org.autojs.autojs.util.RhinoUtils.isMainThread

/**
 * Created by Stardust on May 2, 2017.
 * Transformed by SuperMonster003 on Oct 25, 2023.
 */
class UiHandler(val applicationContext: Context) : Handler(Looper.getMainLooper()) {

    fun toast(scriptRuntime: ScriptRuntime, message: String, isLong: Boolean = false) {
        val showRunnable = Runnable {
            val toastLength = if (isLong) LENGTH_LONG else LENGTH_SHORT
            Toast.makeText(AutoJs.instance.application, message, toastLength)
                .also { ScriptToast.add(it, scriptRuntime) }
                .show()
        }
        when {
            isMainThread() -> showRunnable.run()
            else -> this@UiHandler.post(showRunnable)
        }
    }

    fun dismissAllToasts(scriptRuntime: ScriptRuntime) = ScriptToast.dismissAll(scriptRuntime)

}
